src/main/java/co/cask/gcp/speech/SpeechToTextTransform.java

```diff
@@ -0,0 +1,332 @@
+
+    StructuredRecord.Builder outputBuilder = StructuredRecord.builder(outputRecordSchema);
+
+    List<StructuredRecord> transcriptParts = new ArrayList<>();
+    StringBuilder completeTranscript = new StringBuilder();
+    for (SpeechRecognitionResult result : results) {
+      if (config.getTranscriptionPartsField() != null && !config.getTranscriptionPartsField().isEmpty()) {
+        addTranscriptWithConfidence(outputRecordSchema, transcriptParts, result.getAlternativesList());
+      }
+      if (config.getTranscriptionTextField() != null && !config.getTranscriptionTextField().isEmpty()) {
+        completeTranscript.append(result.getAlternativesList().get(0).getTranscript());
+      }
+    }
+    if (config.getTranscriptionPartsField() != null && !config.getTranscriptionPartsField().isEmpty()) {
+      outputBuilder.set(config.getTranscriptionPartsField(), transcriptParts);
+    }
+    if (config.getTranscriptionTextField() != null && !config.getTranscriptionTextField().isEmpty()) {
+      outputBuilder.set(config.getTranscriptionTextField(), completeTranscript.toString());
+    }
+
+    // copy other schema field values
+    List<Schema.Field> fields = input.getSchema().getFields();
+    if (fields != null) {
+      for (Schema.Field field : fields) {
+        String name = field.getName();
+        if (outputRecordSchema.getField(name) != null) {
+          outputBuilder.set(name, input.get(name));
+        }
+      }
+    }
+    emitter.emit(outputBuilder.build());
+  }
+
+  @Override
+  public void destroy() {
+    super.destroy();
+    try {
+      speech.close();
+    } catch (Exception e) {
+      // no-op
+    }
+  }
+
+  private Schema getSchema(Schema outSchema, @Nullable String transcriptPartsField,
+                           @Nullable String transcriptTextField) {
+    List<Schema.Field> topLevel = new ArrayList<>();
+    boolean hasTranscriptField = false;
+    // one of the output field in which the transcript will be specified must be specified
+    if (transcriptPartsField != null && !transcriptPartsField.isEmpty()) {
+      topLevel.add(transcriptWithConfidenceSchema(transcriptPartsField));
+      hasTranscriptField = true;
+    }
+
+    if (transcriptTextField != null && !transcriptTextField.isEmpty()) {
+      topLevel.add(topConfidenceSchema(transcriptTextField));
+      hasTranscriptField = true;
+    }
+
+    if (!hasTranscriptField) {
+      throw new IllegalArgumentException("Either 'Transcript Parts Field' or 'Transcript Text Field' or both must be " +
+                                           "specified.");
+    }
+
+    if (outSchema != null && outSchema.getFields() != null) {
+      topLevel.addAll(outSchema.getFields());
+    }
+    return Schema.recordOf("record", topLevel);
+  }
+
+  /**
+   * Request object from the Plugin REST call.
+   */
+  public static final class Request extends SpeechTransformConfig {
+    private Map<String, Schema> inputSchemas;
```

Comment:
why is this taking multiple input schemas?

Reply:
I don't think the transform was written initially with much though lot of weird stuff going on.

Fixed this one.